src/wp-includes/link-template.php

```diff
@@ -1895,109 +1920,131 @@ function get_adjacent_post( $in_same_term = false, $excluded_terms = '', $previo
 		 * current user has the 'read_private_posts' cap.
 		 */
 		$private_states = get_post_stati( array( 'private' => true ) );
-		$where         .= " AND ( p.post_status = 'publish'";
+		$args['post_status'] = array( 'publish' );
 		foreach ( $private_states as $state ) {
 			if ( current_user_can( $read_private_cap ) ) {
-				$where .= $wpdb->prepare( ' OR p.post_status = %s', $state );
+				$args['post_status'][] = $state;
 			} else {
-				$where .= $wpdb->prepare( ' OR (p.post_author = %d AND p.post_status = %s)', $user_id, $state );
+				$args['post_status'][] = $state;
+				$args['author__in'][]  = $user_id;
 			}
 		}
```

Comment:
`WP_Query` deals with the logged in user and private status capability checks so you may be able to drop this block.

https://github.com/WordPress/wordpress-develop/blob/3d5454875ba02936d20ef5fc67d5cb55d7eefebe/src/wp-includes/class-wp-query.php#L2632-L2638

Reply:
Good catch. Removing the code, fixes the unit test. Removed now.